src/core/useVideoTexture.tsx

```diff
@@ -1,24 +1,41 @@
 import * as THREE from 'three'
-import { useEffect } from 'react'
+import { useCallback, useEffect, useRef } from 'react'
 import { useThree } from '@react-three/fiber'
-import { suspend, preload, clear } from 'suspend-react'
+import { suspend } from 'suspend-react'
+import Hls, { HlsConfig } from 'hls.js'
```

Comment:
I need to check that this is published correctly since default exports are a general hazard. If so, or it doesn't tree-shake, we'll need to move it to a lazy `import` in the guard you have within `suspend`. The module reference can be memoized within this file.

Reply:
@CodyJasonBennett is it good to go? @netgfx

Reply:
@drcmda I think I need to make sure that HLS gets removed when the component unmounts, it is a bit tricky because when setting ref values from inside the promise those aren't available outside of it like in the useEffect in order to call them on the useEffect return